<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="windowModality">
   <enum>Qt::NonModal</enum>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>659</width>
    <height>542</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>659</width>
    <height>542</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>659</width>
    <height>542</height>
   </size>
  </property>
  <property name="acceptDrops">
   <bool>false</bool>
  </property>
  <property name="windowTitle">
   <string>PlayStation 3 Flash Tool v4.0 (Public Release) By Abkarino (NAND ECC Algo. By RPS)</string>
  </property>
  <property name="windowIcon">
   <iconset resource="ps3flashtool.qrc">
    <normaloff>:/icons/res/PS3NAND.png</normaloff>:/icons/res/PS3NAND.png</iconset>
  </property>
  <property name="autoFillBackground">
   <bool>true</bool>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QGroupBox" name="grbNANDFiles">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>0</y>
      <width>641</width>
      <height>141</height>
     </rect>
    </property>
    <property name="title">
     <string>PS3 Flash Image Files:</string>
    </property>
    <widget class="QCustomLineEdit" name="txtNAND0">
     <property name="geometry">
      <rect>
       <x>144</x>
       <y>20</y>
       <width>440</width>
       <height>20</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="lblNAND0">
     <property name="geometry">
      <rect>
       <x>7</x>
       <y>20</y>
       <width>131</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>NAND 0:</string>
     </property>
     <property name="scaledContents">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QPushButton" name="btnNAND0">
     <property name="geometry">
      <rect>
       <x>590</x>
       <y>20</y>
       <width>40</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>...</string>
     </property>
    </widget>
    <widget class="QCustomLineEdit" name="txtNAND1">
     <property name="geometry">
      <rect>
       <x>144</x>
       <y>49</y>
       <width>440</width>
       <height>20</height>
      </rect>
     </property>
    </widget>
    <widget class="QPushButton" name="btnNAND1">
     <property name="geometry">
      <rect>
       <x>590</x>
       <y>49</y>
       <width>40</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>...</string>
     </property>
    </widget>
    <widget class="QLabel" name="lblNAND1">
     <property name="geometry">
      <rect>
       <x>7</x>
       <y>49</y>
       <width>131</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>NAND 1:</string>
     </property>
    </widget>
    <widget class="QLabel" name="lblNAND">
     <property name="geometry">
      <rect>
       <x>7</x>
       <y>80</y>
       <width>131</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>NAND:</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnNAND">
     <property name="geometry">
      <rect>
       <x>590</x>
       <y>80</y>
       <width>40</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>...</string>
     </property>
    </widget>
    <widget class="QCustomLineEdit" name="txtNAND">
     <property name="geometry">
      <rect>
       <x>144</x>
       <y>80</y>
       <width>440</width>
       <height>20</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="lblPatch">
     <property name="geometry">
      <rect>
       <x>7</x>
       <y>110</y>
       <width>131</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Select Patch Method:</string>
     </property>
    </widget>
    <widget class="QComboBox" name="cboPatches">
     <property name="geometry">
      <rect>
       <x>144</x>
       <y>110</y>
       <width>440</width>
       <height>22</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>Patch dump using Darkjiros 4.70 CFW (NoFSM) patch</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Patch dump using 3.55 CFW (FSM) patch</string>
      </property>
     </item>
    </widget>
   </widget>
   <widget class="QGroupBox" name="grbProcess">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>144</y>
      <width>533</width>
      <height>71</height>
     </rect>
    </property>
    <property name="title">
     <string>Process:</string>
    </property>
    <widget class="QComboBox" name="cboProcess">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>20</y>
       <width>471</width>
       <height>22</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>Automatic patch raw nand flash</string>
      </property>
      <property name="icon">
       <iconset>
        <normalon>:/icons/res/PS3NAND.png</normalon>
       </iconset>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Unscramble &amp; make a unified (interleaved) nand flash dump</string>
      </property>
      <property name="icon">
       <iconset>
        <normalon>:/icons/res/PS3NAND.png</normalon>
       </iconset>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Rescramble &amp; make a seprate (deInterleaved) nand flash dumps</string>
      </property>
      <property name="icon">
       <iconset>
        <normalon>:/icons/res/PS3NAND.png</normalon>
       </iconset>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Extract nand flash dump files</string>
      </property>
      <property name="icon">
       <iconset>
        <normalon>:/icons/res/PS3NAND.png</normalon>
       </iconset>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Patch interleaved nand flash dump</string>
      </property>
      <property name="icon">
       <iconset>
        <normalon>:/icons/res/PS3NAND.png</normalon>
       </iconset>
      </property>
     </item>
    </widget>
    <widget class="QLabel" name="lblProcess">
     <property name="geometry">
      <rect>
       <x>3</x>
       <y>20</y>
       <width>41</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Process:</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="chkDiffFile">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>46</y>
       <width>131</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>Create differential file.</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QCheckBox" name="chkExtractConsoleData">
     <property name="geometry">
      <rect>
       <x>200</x>
       <y>46</y>
       <width>151</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>Extract console data files.</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QCheckBox" name="chkKeepUnifiedDump">
     <property name="geometry">
      <rect>
       <x>350</x>
       <y>46</y>
       <width>121</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>Keep unified dump.</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="grbOperation">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>144</y>
      <width>101</width>
      <height>71</height>
     </rect>
    </property>
    <property name="title">
     <string>Operation:</string>
    </property>
    <widget class="QPushButton" name="btnStart">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>19</y>
       <width>81</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>&amp;Start</string>
     </property>
     <property name="flat">
      <bool>false</bool>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="grbProcessLog">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>216</y>
      <width>641</width>
      <height>251</height>
     </rect>
    </property>
    <property name="title">
     <string>Process Log:</string>
    </property>
    <widget class="QTextEdit" name="txtProcessLog">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>24</y>
       <width>551</width>
       <height>220</height>
      </rect>
     </property>
    </widget>
    <widget class="QPushButton" name="btnClearLog">
     <property name="geometry">
      <rect>
       <x>570</x>
       <y>23</y>
       <width>61</width>
       <height>220</height>
      </rect>
     </property>
     <property name="text">
      <string>&amp;Clear Log</string>
     </property>
     <property name="flat">
      <bool>false</bool>
     </property>
    </widget>
   </widget>
   <widget class="QProgressBar" name="proProcess">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>474</y>
      <width>641</width>
      <height>23</height>
     </rect>
    </property>
    <property name="value">
     <number>0</number>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>659</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_File">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="mnuLoadNAND0Dump"/>
    <addaction name="mnuLoadNAND1Dump"/>
    <addaction name="separator"/>
    <addaction name="mnuSaveInterleavedNANDDump"/>
    <addaction name="mnuLoadInterleavedNANDDump"/>
    <addaction name="separator"/>
    <addaction name="mnuExit"/>
   </widget>
   <widget class="QMenu" name="menu_Process">
    <property name="title">
     <string>&amp;Process</string>
    </property>
    <addaction name="mnuAutomatic_patch_raw_nand_flash"/>
    <addaction name="separator"/>
    <addaction name="mnuUnscramble"/>
    <addaction name="mnuRescramble"/>
    <addaction name="separator"/>
    <addaction name="mnuExtract"/>
    <addaction name="mnuPatch"/>
   </widget>
   <widget class="QMenu" name="menu_Help">
    <property name="title">
     <string>&amp;Help</string>
    </property>
    <addaction name="action_Help"/>
    <addaction name="separator"/>
    <addaction name="mnuAbout"/>
   </widget>
   <addaction name="menu_File"/>
   <addaction name="menu_Process"/>
   <addaction name="menu_Help"/>
  </widget>
  <widget class="QStatusBar" name="stbMain">
   <property name="minimumSize">
    <size>
     <width>659</width>
     <height>20</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>659</width>
     <height>20</height>
    </size>
   </property>
   <property name="sizeGripEnabled">
    <bool>false</bool>
   </property>
  </widget>
  <action name="mnuLoadNAND0Dump">
   <property name="text">
    <string>Load NAND &amp;0 Dump</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+0</string>
   </property>
  </action>
  <action name="mnuLoadNAND1Dump">
   <property name="text">
    <string>Load NAND &amp;1 Dump</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+1</string>
   </property>
  </action>
  <action name="mnuSaveInterleavedNANDDump">
   <property name="text">
    <string>&amp;Save Interleaved NAND Dump</string>
   </property>
  </action>
  <action name="mnuLoadInterleavedNANDDump">
   <property name="text">
    <string>Load Interleaved NAND Dump</string>
   </property>
  </action>
  <action name="actionLoad_Donor_Interleaved_NAND_Dump">
   <property name="text">
    <string>Load &amp;Donor Interleaved NAND Dump</string>
   </property>
  </action>
  <action name="mnuExit">
   <property name="text">
    <string>&amp;Exit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+X</string>
   </property>
  </action>
  <action name="mnuAbout">
   <property name="text">
    <string>&amp;About</string>
   </property>
  </action>
  <action name="action_Help">
   <property name="text">
    <string>&amp;Help</string>
   </property>
  </action>
  <action name="mnuAutomatic_patch_raw_nand_flash">
   <property name="text">
    <string>Automatic patch raw nand flash</string>
   </property>
  </action>
  <action name="mnuUnscramble">
   <property name="text">
    <string>Unscramble &amp;&amp; make a unified (interleaved) nand flash dump</string>
   </property>
  </action>
  <action name="mnuRescramble">
   <property name="text">
    <string>Rescramble &amp;&amp; make a seprate (deInterleaved) nand flash dumps</string>
   </property>
  </action>
  <action name="mnuExtract">
   <property name="text">
    <string>Extract nand flash dump files</string>
   </property>
  </action>
  <action name="mnuPatch">
   <property name="text">
    <string>Patch interleaved nand flash dump</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QCustomLineEdit</class>
   <extends>QLineEdit</extends>
   <header>qcustomlineedit.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="ps3flashtool.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>mnuLoadNAND1Dump</sender>
   <signal>triggered()</signal>
   <receiver>btnNAND1</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>619</x>
     <y>79</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mnuLoadNAND0Dump</sender>
   <signal>triggered()</signal>
   <receiver>btnNAND0</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>619</x>
     <y>50</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mnuExit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>329</x>
     <y>270</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mnuSaveInterleavedNANDDump</sender>
   <signal>triggered()</signal>
   <receiver>btnNAND</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>619</x>
     <y>110</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mnuLoadInterleavedNANDDump</sender>
   <signal>triggered()</signal>
   <receiver>btnNAND</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>619</x>
     <y>110</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mnuAutomatic_patch_raw_nand_flash</sender>
   <signal>triggered()</signal>
   <receiver>btnStart</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>600</x>
     <y>204</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mnuUnscramble</sender>
   <signal>triggered()</signal>
   <receiver>btnStart</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>600</x>
     <y>204</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mnuRescramble</sender>
   <signal>triggered()</signal>
   <receiver>btnStart</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>600</x>
     <y>204</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mnuPatch</sender>
   <signal>triggered()</signal>
   <receiver>btnStart</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>600</x>
     <y>204</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mnuExtract</sender>
   <signal>triggered()</signal>
   <receiver>btnStart</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>600</x>
     <y>204</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
